src/components/Guilds/ActionsModal/index.tsx

```diff
@@ -73,11 +75,41 @@ const ActionModal: React.FC<ActionModalProps> = ({
 
   function getContent() {
     if (selectedFunction) {
+      const contractInterface = new utils.Interface(
+        selectedContract.functions.map(f => {
+          const name = f.functionName;
+          const params = f.params.reduce(
+            (acc, cur, i) =>
+              acc.concat(

```

Comment:
We could maybe make this into a function or hook, I see its used in a few places and a little difficult to understand